client/i18n/schema-validation.ts

```diff
@@ -80,16 +88,22 @@ const findExtraneousKeys = (file, schema, path) => {
  * @param {Object} obj The object to check the values of
  * @param {String} namespace String for tracking nested properties
  */
-const noEmptyObjectValues = (obj, namespace = '') => {
+const noEmptyObjectValues = (
+  obj: Record<string, object | string>,
+  namespace = ''
+): string[] => {
   const emptyKeys = [];
   for (const key of Object.keys(obj)) {
     if (Array.isArray(obj[key])) {
-      if (!obj[key].length) {
+      if (!(obj[key] as string[]).length) {
```

Comment:
We should not have this assertion here. This is not making use of TypeScript.

Instead, a function like:

```ts
function isStringArray(obj: unknown): obj is string[] {}
````

Can be written, and then TS will know its type.

Reply:
Thanks, I learned the proper way to use TS.

But I think this judgment is used to judge the array is empty, maybe the following method is more appropriate, 
```typescript
const isEmptyArray = (obj: unknown) => {
  return Array.isArray(obj) && obj.length === 0;
};
```
And it will be like

```diff
    if (Array.isArray(obj[key])) {
-      if (!obj[key].length) {
+      if (isEmptyArray(obj[key])) {
```
what do you think?

Reply:
The issue is you need to tell TypeScript it is ok to access the `.length` property of some object. So, you need to assert it is an array, before trying to do so.

Reply:
Right, but this does not assert the obj is a `string[]`, just that it is an `Array`

Reply:
Done.
I changed to `obj is unknown[]`, then I thought it may need to be moved to `client/utils/index.ts` later